samples/SocialSample/Startup.cs

```diff
@@ -281,24 +427,46 @@ public void Configure(IApplicationBuilder app)
                 }
 
                 // Display user information
-                context.Response.ContentType = "text/html";
-                await context.Response.WriteAsync("<html><body>");
-                await context.Response.WriteAsync("Hello " + (context.User.Identity.Name ?? "anonymous") + "<br>");
+                var response = context.Response;
+                response.ContentType = "text/html";
+                await response.WriteAsync("<html><body>");
+                await response.WriteAsync("Hello " + (context.User.Identity.Name ?? "anonymous") + "<br>");
                 foreach (var claim in context.User.Claims)
                 {
-                    await context.Response.WriteAsync(claim.Type + ": " + claim.Value + "<br>");
+                    await response.WriteAsync(claim.Type + ": " + claim.Value + "<br>");
                 }
 
-                await context.Response.WriteAsync("Tokens:<br>");
+                await response.WriteAsync("Tokens:<br>");
                 
-                await context.Response.WriteAsync("Access Token: " + await context.GetTokenAsync("access_token") + "<br>");
-                await context.Response.WriteAsync("Refresh Token: " + await context.GetTokenAsync("refresh_token") + "<br>");
-                await context.Response.WriteAsync("Token Type: " + await context.GetTokenAsync("token_type") + "<br>");
-                await context.Response.WriteAsync("expires_at: " + await context.GetTokenAsync("expires_at") + "<br>");
-                await context.Response.WriteAsync("<a href=\"/logout\">Logout</a><br>");
-                await context.Response.WriteAsync("</body></html>");
+                await response.WriteAsync("Access Token: " + await context.GetTokenAsync("access_token") + "<br>");
+                await response.WriteAsync("Refresh Token: " + await context.GetTokenAsync("refresh_token") + "<br>");
+                await response.WriteAsync("Token Type: " + await context.GetTokenAsync("token_type") + "<br>");
+                await response.WriteAsync("expires_at: " + await context.GetTokenAsync("expires_at") + "<br>");
+                await response.WriteAsync("<a href=\"/logout\">Logout</a><br>");
+                await response.WriteAsync("<a href=\"/refresh_token\">Refresh Token</a><br>");
+                await response.WriteAsync("</body></html>");
             });
         }
+
+        private async Task<OAuthOptions> GetOAuthOptionsAsync(IAuthenticationHandlerProvider authSchemes, HttpContext context, string currentAuthType)
+        {
+            var handler = await authSchemes.GetHandlerAsync(context, currentAuthType);
+
+            if (string.Equals(GoogleDefaults.AuthenticationScheme, currentAuthType))
+            {
+                return ((AuthenticationHandler<GoogleOptions>)handler).Options;
```

Comment:
Since you already have a switch you should just do `IOptionsMonitor<GoogleOptions>.Get(GoogleDefaults.AuthenticationScheme)` and skip resolving the handler, same for the other cases